File: server/Models/Client/ResidentialClient.cs
namespace server.Models.Client;

public class ResidentialClient
{
  public int Id { get; set; }
  public int Uid { get; set; }
  public Client Client { get; set; } = null!;
  public string FirstName { get; set; } = string.Empty;
  public string LastName { get; set; } = string.Empty;
  public string Email { get; set; } = string.Empty;
  public string PhoneNumber { get; set; } = string.Empty;
  public Address Address { get; set; } = new Address();
}

File: client/src/types/ResidentialClient.ts
type ResidentialClient = {
  uid: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  notes?: string;
  createdAt: string;
  address: {
    street: string;
    unit?: string;
    city: string;
    state: string;
    zipCode: string;
  };
};

type CreateResidentialClient = {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  notes?: string;
  address: {
    street: string;
    unit?: string;
    city: string;
    state: string;
    zipCode: string;
  };
};

export type { ResidentialClient, CreateResidentialClient };
